mooncake-store/src/hf3fs/hf3fs_file.cpp

```
@@ -0,0 +1,355 @@
+    }
+
+    // 2. Get thread resources
+    auto* resource = resource_manager_->getThreadResource();
+    if (!resource || !resource->initialized) {
+        return make_error<size_t>(ErrorCode::FILE_OPEN_FAIL);
+    }
+
+    // 3. Acquire write lock
+    auto lock = acquire_write_lock();
+    if (!lock.is_locked()) {
+        return make_error<size_t>(ErrorCode::FILE_LOCK_FAIL);
+    }
+
+    // 4. Write in chunks
+    auto& threefs_iov = resource->iov_;
+    auto& ior_write = resource->ior_write_;
+    const char* data_ptr = data.data();
+    size_t total_bytes_written = 0;
+    off_t current_offset = 0;
+    const size_t max_chunk_size = resource->config_.iov_size;
+
+    while (total_bytes_written < length) {
+        // Calculate current chunk size
+        size_t chunk_size = std::min(length - total_bytes_written, max_chunk_size);
+
+        // Copy data to shared buffer
+        memcpy(threefs_iov.base, data_ptr + total_bytes_written, chunk_size);
+
+        // Prepare IO request
+        int ret = hf3fs_prep_io(&ior_write, &threefs_iov, false,
+                               threefs_iov.base, fd_, current_offset, chunk_size, nullptr);
+        if (ret < 0) {
+            return make_error<size_t>(ErrorCode::FILE_WRITE_FAIL);
+        }
+
+        // Submit IO request
+        ret = hf3fs_submit_ios(&ior_write);
+        if (ret < 0) {
+            return make_error<size_t>(ErrorCode::FILE_WRITE_FAIL);
+        }
+
+        // Wait for IO completion
+        struct hf3fs_cqe cqe;
+        ret = hf3fs_wait_for_ios(&ior_write, &cqe, 1, 1, nullptr);
+        if (ret < 0 || cqe.result < 0) {
+            return make_error<size_t>(ErrorCode::FILE_WRITE_FAIL);
+        }
+
+        size_t bytes_written = cqe.result;
+        total_bytes_written += bytes_written;
+        current_offset += bytes_written;
+
+        if (bytes_written < chunk_size) {
+            break; // Short write, possibly disk full
```

We should return a error?

---

This problem will be checked in `total_bytes_read != length` , so we don't need to return error here.